zap/src/config.rs

```diff
@@ -141,22 +192,41 @@ pub enum Enum<'src> {
 }
 
 impl<'src> Enum<'src> {
-	pub fn max_size(&self, config: &Config<'src>, recursed: &mut HashSet<&'src str>) -> Option<usize> {
+	pub fn size(
+		&self,
+		tydecls: &HashMap<&'src str, TyDecl<'src>>,
+		recursed: &mut HashSet<&'src str>,
+	) -> (usize, Option<usize>) {
 		match self {
-			Self::Unit(vec) => Some(NumTy::from_f64(0.0, vec.len() as f64).size()),
+			Self::Unit(enumerators) => {
+				let numty = NumTy::from_f64(0.0, enumerators.len() as f64);
+
+				(numty.min() as usize, Some(numty.max() as usize))
+			}
 
 			Self::Tagged { variants, .. } => {
-				let mut size = NumTy::from_f64(0.0, variants.len() as f64).size();
+				let mut min = 0;
+				let mut max = Some(0);
+
+				for (_, ty) in variants.iter() {
+					let (ty_min, ty_max) = ty.size(tydecls, recursed);
+
+					if ty_min < min {
+						min = ty_min;
+					}
 
-				for (_, ty) in variants {
-					if let Some(ty_size) = ty.max_size(config, recursed) {
-						size += ty_size;
+					if let Some(ty_max) = ty_max {
+						if let Some(current_max) = max {
+							if ty_max > current_max {
+								max = Some(ty_max);
+							}
+						}
 					} else {
-						return None;
+						max = None;
```

Comment:
Setting this to max may overwrite the previously defined Max type. Not sure if it's intended.